File: src/main/java/cz/muni/fi/pa039/widgets/AbstractWidget.java
package cz.muni.fi.pa039.widgets;

public abstract class AbstractWidget {
    private boolean selected = false;

    public abstract boolean pointIsInside(int x, int y);
    public abstract int getId();

    public int getStroke() {
        return 0;
    }

    public int getFill() {
        return selected ? 230 : 255;
    }

    public boolean isSelected() {
        return selected;
    }

    public void setSelected(boolean selected) {
        this.selected = selected;
    }
}

File: src/cz/muni/fi/pa093/widgets/AbstractWidget.kt
package cz.muni.fi.pa093.widgets

interface AbstractWidget {
    var isSelected: Boolean
    val id: Int
    val stroke: Int
        get() = 0

    val fill: Int
        get() = if (isSelected) 230 else 255

    fun pointIsInside(x: Int, y: Int): Boolean
}
